### PR TITLE
Non-GitPy GitRepo.get_branch_commits()

### PR DESCRIPTION
**Pull request by mih**
This is sitting on top of gh-3833. And aims to be another step towards #2970 by fixing #3813

Sibling PR in the crawler is https://github.com/datalad/datalad-crawler/pull/66

### CODE REVIEW

**Comment by bpoldrack** [comment]
LGTM

**Comment by mih** [comment]
I prepared an adjustment PR for the crawler https://github.com/datalad/datalad-crawler/pull/66

I would merge this when the test have gone through.

File: datalad/tests/utils.py [hunk 1]
```diff
@@ -1827,6 +1827,37 @@ def newfunc(*args, **kwargs):
     return newfunc
 
 
+def get_branch_commits(repo, branch=None, limit=None, stop=None):
+    """Return commit hexshas for a branch
```
Comment (yarikoptic) [inline datalad/tests/utils.py]: My main question is - why not to keep it as a method (well, rename to `get_branch_commits_`) of GitRepo and just remove `value` parameter, thus always yielding `hexsha`s?  
or we do not foresee any need for such functionality for anything but the tests?  I already see one usecase RFed above in https://github.com/datalad/datalad/pull/3834/files#diff-18445826672fc8f047d0b7d56512986cR327 for direct call to `git` (IMHO suboptimal to breed those)

Comment (mih) [reply datalad/tests/utils.py]: The idea here, and in other PRs before, is to get rid all these tiny wrapper around simple git calls and leave them to the return value focused wrappers we have now. I was under the impression that this direction had received general support. I dont see why this particular method should be treated differently.

Comment (mih) [reply datalad/tests/utils.py]: I have now implemented what you suggested @yarikoptic and adjusted the companion PR in the crawler too.

**Comment by mih** [comment]
I have dropped all changes that removed this method from the class and turned into a standalone function.
Remaining changes are:

- rename to match generator-method convention
- stripping the flexibility of the method down to what is needed in core and crawler, and reimplement it with `rev-list`
- adjust usage due to those changes

**Comment by mih** [comment]
As with other PRs the windows github ci failure is unrelated. Crawler failure is addressed in companion PR.

**Pull request merged**